sdk/python/feast/feature_store.py

```diff
@@ -1018,6 +1018,30 @@ def _augment_response_with_on_demand_transforms(
                     ] = GetOnlineFeaturesResponse.FieldStatus.PRESENT
         return OnlineResponse(GetOnlineFeaturesResponse(field_values=result_rows))
 
+    def _get_feature_views_to_use(
+        self,
+        features: Optional[Union[List[str], FeatureService]],
+        allow_cache=False,
+        hide_dummy_entity: bool = True,
+    ) -> List[FeatureView]:
+
+        all_feature_views = {
+            fv.name: fv
+            for fv in self._list_feature_views(allow_cache, hide_dummy_entity)
+        }
+
+        if isinstance(features, FeatureService):
+            for fv in features.feature_views:
+                if fv.name not in all_feature_views:
+                    raise ValueError(
+                        f"{fv.name} used in the FeatureService is not in the registry."
```

Comment:
We're trying to make our exceptions clearer for users. If I read an exception like this it isn't really clear to me what to do as a beginner. What about

Something like
`"The provided feature service {name} contains a reference to a feature view {fv_name} which doesn't exist. Please make sure that you have created the feature view {fv_name} and that you have registered it by running \"apply\""`

Reply:
sounds good